### ISSUE TITLE
Windows install configuration defaults to debug

### ISSUE THREAD

**Issue by burgholzer** [scikit-build/scikit-build-core#629]
Hey 👋🏼 

I noticed some unexpected behavior while working on adding proper installation instructions for one of our C++-based projects (https://github.com/cda-tum/mqt-core) that uses scikit-build-core and pybind11 to expose the project to Python.
We use cibuildwheel to build binary wheels for all possible platforms, including Windows.
Due to better support for caching (via ccache) we want to use Ninja for the build also on Windows (which would normally default to using the MSVC generator). Thus, we explicitly set the `CMAKE_GENERATOR` in the pyproject.toml configuration of cibuildwheel (see https://github.com/cda-tum/mqt-core/blob/3ee8bea3ef9ff22b849cb11d811e4641692459ef/pyproject.toml#L278-L281).
This is working fine in principle. See https://github.com/cda-tum/mqt-core/actions/runs/7678858842/job/20929146794 for a successful example run.
- ✅ It uses Ninja: https://github.com/cda-tum/mqt-core/actions/runs/7678858842/job/20929146794#step:7:333
  ```
  -- Building for: Ninja
  ```
- ✅ It uses MSVC as a compiler: https://github.com/cda-tum/mqt-core/actions/runs/7678858842/job/20929146794#step:7:334
  ```
  The CXX compiler identification is MSVC 19.37.32826.1
  ```
- ❌ It uses "Debug" for the install configuration: https://github.com/cda-tum/mqt-core/actions/runs/7678858842/job/20929146794#step:7:486
  ```
  *** Installing project into wheel...
  -- Install configuration: "Debug"
  ```

Although the default `cmake.build-type = "Release"` and I would have assumed the install type to be the same.
Maybe this is just a faulty assumption. If so, is there any way that I could properly set the install configuration?
If not, could this point to a potential bug within scikit-build-core? Maybe a missing `--config cmake.build_type` somewhere?

The above scenario is running under `scikit-build-core 0.8.0 using CMake 3.27.9 (wheel)`.

**Comment by henryiii** [comment]
We pick up the generator here: https://github.com/scikit-build/scikit-build-core/blob/2eb038e74fd5afdfb99e9b9d638e232c4cde37c9/src/scikit_build_core/cmake.py#L202

Then we set multiple config here: https://github.com/scikit-build/scikit-build-core/blob/2eb038e74fd5afdfb99e9b9d638e232c4cde37c9/src/scikit_build_core/cmake.py#L215

And then add the --config here: https://github.com/scikit-build/scikit-build-core/blob/2eb038e74fd5afdfb99e9b9d638e232c4cde37c9/src/scikit_build_core/cmake.py#L260-L261

I've suspected our Ninja on Windows support is a little shaky, could very well be on to something. Just don't see where it's breaking yet. Maybe you could increase verbosity and rerun?

```toml
[tool.scikit-build]
cmake.verbose = true
logging.level = "INFO"
```

**Comment by henryiii** [comment]
Oh, you are using the single-config generator for Ninja, not the multi-config one, so that should be set the normal way, not passed to --install, I believe.

**Comment by burgholzer** [comment]
> I've suspected our Ninja on Windows support is a little shaky, could very well be on to something. Just don't see where it's breaking yet. Maybe you could increase verbosity and rerun?
> 
> ```toml
> [tool.scikit-build]
> cmake.verbose = true
> logging.level = "INFO"
> ```

Sure. Here is the link to the respective workflow run: https://github.com/cda-tum/mqt-core/actions/runs/7898432876/job/21555816974
Maybe that provides some more information.

**Comment by henryiii** [comment]
Ahh, apologies, once more: could you change `logging.level = "INFO"` to `logging.level = "DEBUG"`? The run commands are only printed out at the debug level.

**Comment by burgholzer** [comment]
> Ahh, apologies, once more: could you change `logging.level = "INFO"` to `logging.level = "DEBUG"`? The run commands are only printed out at the debug level.

No worries. A corresponding run can be found here: https://github.com/cda-tum/mqt-core/actions/runs/7903453484/job/21571390390

**Comment by LecrisUT** [comment]
I think the issue is that `build_type` is not evaluated even though it should be defaulted to `Release`, can you add that manually?

`single_config` seems to be set since I don't see any `--config` being passed

**Comment by burgholzer** [comment]
> I think the issue is that `build_type` is not evaluated even though it should be defaulted to `Release`, can you add that manually?

Sure (explicitly set in https://github.com/cda-tum/mqt-core/commit/fcc5d095da2eef1a1c5e2a1facff60889918179d), see https://github.com/cda-tum/mqt-core/actions/runs/7904628863/job/21575380948 for the corresponding workflow run.
Doesn't seem to affect the install configuration.

**Comment by henryiii** [comment]
The build type defaults to `Release`, though it supports being set to an empty string (which in CMake is for packager controlled settings), so setting it to Release shouldn't affect anything. I don't see any mention of DCMAKE_BUILD_TYPE in the logs, which sounds like there is a logic bug somewhere.

**Comment by henryiii** [comment]
Ah, I think the order is an issue. This is getting set after the command line is produces (since it's looking for `-G` options), but that means it can't affect the produced command line!

**Comment by henryiii** [comment]
Great, thanks! (You can also see this is now in the configure command it runs) Will release a patch release soon.

**Comment by burgholzer** [comment]
Many thanks for the quick fix!

**Comment by henryiii** [comment]
Released 0.8.1.
